### ISSUE TITLE
Babel react modules deleted by CK3?

### ISSUE THREAD

**Issue by robstown** [bdkjones/CodeKit#200]
**Quick, short summary:**
Transpiling with babel was working fine until it wasn't.  After a save of a file, I started getting following errors everytime I tried to save a file with Babel transpile option:

```
Babel: Transpiling failed:

ReferenceError: [BABEL] /Users/Rob/Robs-Stuff/RobUniversity/ProgrammingPlaybox/mobx-react-todomvc/src/components/todoApp.js: Unknown option: foreign.Children. Check out http://babeljs.io/docs/usage/options/ for more info
    at Logger.error (/Applications/CodeKit.app/Contents/Resources/engines/node_modules/babel-core/lib/transformation/file/logger.js:41:11)
    at OptionManager.mergeOptions (/Applications/CodeKit.app/Contents/Resources/engines/node_modules/babel-core/lib/transformation/file/options/option-manager.js:220:20)
    at /Applications/CodeKit.app/Contents/Resources/engines/node_modules/babel-core/lib/transformation/file/options/option-manager.js:259:14
    at /Applications/CodeKit.app/Contents/Resources/engines/node_modules/babel-core/lib/transformation/file/options/option-manager.js:312:20
    at Array.map (native)
    at OptionManager.resolvePresets (/Applications/CodeKit.app/Contents/Resources/engines/node_modules/babel-core/lib/transformation/file/options/option-manager.js:269:20)
    at OptionManager.mergePresets (/Applications/CodeKit.app/Contents/Resources/engines/node_modules/babel-core/lib/transformation/file/options/option-manager.js:258:10)
    at OptionManager.mergeOptions (/Applications/CodeKit.app/Contents/Resources/engines/node_modules/babel-core/lib/transformation/file/options/option-manager.js:243:14)
    at OptionManager.init (/Applications/CodeKit.app/Contents/Resources/engines/node_modules/babel-core/lib/transformation/file/options/option-manager.js:351:12)
    at File.initOptions (/Applications/CodeKit.app/Contents/Resources/engines/node_modules/babel-core/lib/transformation/file/index.js:216:65)
```

It appears maybe one of the babel plugins for react is no longer there?  If I uncheck react preset, I don't get the error (but I do get the error saying my code has errors in the jsx sections).  Is it possible that CK3 deleted the file?  I am not doing anything in that directory, so I can't see how I would have deleted it on accident.

**Expected results:**
Transpiles normally

**Actual results:**
error

**Exact steps to reproduce:**
try to transpile using Babel and react preset

**A link to download a simplified project or file that shows the issue:**
can provide if needed, let me know what you need

**Your configuration (any details about your system that you think might be relevant)**
CK3.0 build 25288

**Comment by imtiazwazir** [comment]
Same problem here and it occurs after I changed the build output file name

<img width="502" alt="screen shot 2016-11-01 at 5 05 18 pm" src="https://cloud.githubusercontent.com/assets/1964969/19889197/72db3b34-a055-11e6-89b1-1e2c5217b4ba.png">

**Comment by bdkjones** [comment]
Can one of you upload a simple demo project showing this issue? Install whatever's needed to produce the conflict so I can take a look.

**Comment by imtiazwazir** [comment]
[Archive.zip](https://github.com/bdkjones/CodeKit/files/566376/Archive.zip)

please find attached

**Comment by XantheStudios** [comment]
Hey @robstown and @imtiazwazir. Just wondering of either have you been successful with th e transpiling  or linting of a React project. If yes, would appreciate any direction on achieving this myself. Cheers.

**Comment by imtiazwazir** [comment]
@bdkjones is busy making more jokes about Apple, I have dropped my all hopes about this issue :)

**Comment by bdkjones** [comment]
I haven't ignored it; it's just that this is a SUPER RARE edge case and a workaround has been found. Development cycles are hard to come by with a one-man shop, so I have to prioritize things that are bigger issues

**Comment by robstown** [comment]
Hey @XantheStudios , sorry for long delay, I was out of commission for several months.  To answer you question, the only work-around I know is to load all of the babel npm modules local to your project.  This sucks but I don't know any other way around it....

**Comment by sami616** [comment]
wow this still seems to be an issue.... bummer. back to webpack i guess.

**Comment by bdkjones** [comment]
So I *think* I understand what's happening here. Babel allows you to abbreviate the plugin names you pass to it so that these are equivalent:

    babel-preset-react
    react

If the `babel-preset` is missing from the name that you pass in, Babel adds it and then searches for that. However, I'm not sure that behavior is working correctly since I extend Babel to search for globally installed presets/plugins. 

I will change CodeKit to always pass the long-form name of the plugin/preset for react and maybe that will solve this?

**Comment by bdkjones** [comment]
Yep. That appears to be the issue. Changing CodeKit to always supply Babel with the "longform" names seems to resolve the conflict so that the `react` npm module is not mistakenly identified as `babel-preset-react` to Babel. 

This is why we pick one name and use it. My name is `Bryan`. You cannot call me `yan` and expect me to respond.

**Comment by bdkjones** [comment]
Fix implemented for 3.3.
